pennylane/labs/trotter/realspace/realspace_coefficients.py

```diff
@@ -0,0 +1,346 @@
+"""Tree representation of coefficients of a realspace operator"""
+
+from __future__ import annotations
+
+from enum import Enum
+from itertools import product
+from typing import Any, Iterator, Tuple, Union
+
+import numpy as np
+from numpy import allclose, isclose, ndarray, zeros
+
+# pylint: disable=too-many-arguments,too-many-positional-arguments,protected-access,too-many-return-statements
+
+
+class NodeType(Enum):
```

Comment:
Please remember that this will appear in the docs. Better to consider another mechanism, e.g., dictionary etc.